src/ship.js

```diff
@@ -0,0 +1,84 @@
+'use strict'
+
+const settings = {
+    defaults: {
+        _isInitialized: false,
+        _currentPort: null
+    },
+    errors: {
+        alreadyInitialized: 'The ship has already been initialized.',
+        emptyItinerary: 'The ship has no itinerary.',
+        alreadySailing: 'The ship has already set sail.',
+        hasNotPreviouslyDocked: 'You must set sail before you can dock.',
+        alreadyDocked: 'The ship is already docked.'
+    }
+};
+
+class Ship {
+    constructor({ _itinerary }) {
+        this._isInitialized = settings.defaults._isInitialized;
+        this._itinerary = _itinerary;
+        this._currentPort = settings.defaults._currentPort;
+        this._previouslyDockedPorts = [];
+        this._init();
+    }
+
+    _init() {
+        if (this._isInitialized) {
+            throw new Error(settings.errors.alreadyInitialized);
+        }
+
+        if (this._itinerary.ports.length) {
+            this._currentPort = this._itinerary.ports[0];
+            this._currentPort.addShip(this);
+            this._isInitialized = true;
+        }
+    }
+
+    _setSail() {
+        const emptyItinerary = !this._itinerary.ports.length;
+        const hasPreviouslyDocked = this._previouslyDockedPorts.length > 0;
+        const finalDestination = this._itinerary.ports.length === 1;
+        const alreadySailing = !this._currentPort || hasPreviouslyDocked && finalDestination;
```

Comment:
Hmm I see, I had done it like that because I thought it would improve the readability, but I do agree with what you're saying, I was a bit unsure whether to create them as variables or not, but the sole reason was for readability, I will remove them however, I did think it added unnecessary bulk but was hoping it would come up in the code review to help clear it up for me :D